web/skins/classic/views/js/watch.js

```diff
@@ -593,27 +601,105 @@ function fetchImage(streamImage) {
+  coordinateMouse.start_y = y;
+}
+
+function updatePrevCoordinatFrame(x, y) { //Update the Frame's current coordinates
+  PrevCoordinatFrame.x = x;
+  PrevCoordinatFrame.y = y;
+}
+
+function getCoordinateMouse(event) { //We get the current cursor coordinates taking into account the scale relative to the frame size.
   const target = $j(event.target);
-  const width = target.width();
-  const height = target.height();
 
-  const scaleX = parseFloat(monitorWidth / width);
-  const scaleY = parseFloat(monitorHeight / height);
+  const scaleX = parseFloat(monitorWidth / target.width());
+  const scaleY = parseFloat(monitorHeight / target.height());
   const pos = target.offset();
-  const x = parseInt((event.pageX - pos.left) * scaleX);
-  const y = parseInt((event.pageY - pos.top) * scaleY);
-
-  if (showMode == 'events' || !imageControlMode) {
-    if ( event.shift || event.shiftKey ) {
-      streamCmdPan(x, y);
-    } else if (event.ctrlKey) {
-      streamCmdZoomOut();
-    } else {
-      streamCmdZoomIn(x, y);
-    }
+
+  return {x: parseInt((event.pageX - pos.left) * scaleX), y: parseInt((event.pageY - pos.top) * scaleY)}; //The point of the mouse click relative to the dimensions of the real frame.
+}
+
+function handleMove(event) {
+  if (event.ctrlKey && event.shiftKey) {
+    document.ondragstart = function() {
+      return false;
+    }; //Allow drag and drop
   } else {
-    controlCmdImage(x, y);
+    document.ondragstart = function() {}; //Prevent drag and drop
+    return false;
+  }
+
+  if (leftBtnStatus.Down) { //The left button was previously pressed and is now being held. Processing movement with a pressed button.
+    var {x, y} = getCoordinateMouse(event);
+    const k = Math.log(2.72) / Math.log(parseFloat($j('#zoomValue'+monitorId).html())) - 0.3; //Necessary for correctly shifting the image in accordance with the scaling proportions
+
+    coordinateMouse.shiftMouse_x = parseInt((x - coordinateMouse.start_x) * k);
+    coordinateMouse.shiftMouse_y = parseInt((y - coordinateMouse.start_y) * k);
+
+    coordinateMouse.shiftMouseForTrigger_x = Math.abs(parseInt(x - coordinateMouse.start_x));
+    coordinateMouse.shiftMouseForTrigger_y = Math.abs(parseInt(y - coordinateMouse.start_y));
+  }
+  if (event.buttons == 1 && leftBtnStatus.Down != true) { //Start of pressing left button
+    const {x, y} = getCoordinateMouse(event);
+
+    updateCoordinateMouse(x, y);
+    leftBtnStatus.Down = true;
+  } else if (event.buttons == 0 && leftBtnStatus.Down == true) { //Up left button after pressed
+    leftBtnStatus.Down = false;
+    leftBtnStatus.UpAfterDown = true;
+  }
+
+  if ((leftBtnStatus.UpAfterDown) || //The left button was raised or the cursor was moved more than 30 pixels relative to the actual size of the image
+    ((coordinateMouse.shiftMouseForTrigger_x > 30) && leftBtnStatus.Down) ||
+    ((coordinateMouse.shiftMouseForTrigger_y > 30) && leftBtnStatus.Down)) {
+    //We perform frame shift
```

Comment:
The video frame is shifted at intervals of 30 pixels.
This seemed to me the best option, because... if the value is reduced, the load increases and the shift may not keep up with the position of the mouse; if the value is increased, movement is less intuitive.